nilearn/plotting/surf_plotting.py

```diff
@@ -422,6 +426,248 @@ def plot_surf_stat_map(surf_mesh, stat_map, bg_map=None,
     return display
 
 
+def _check_hemispheres(hemispheres):
+    """Checks whether the hemisphere in plot_img_on_surf is correct.
+
+    hemisphere: list
+        Any combination of 'left' and 'right'
+    """
+    if not isinstance(hemispheres, list):
+        hemispheres = [hemispheres]
+    invalid_hemi = any([hemi not in VALID_HEMISPHERES for hemi in hemispheres])
+    if invalid_hemi:
+        supported = "Supported hemispheres:\n" + str(VALID_HEMISPHERES)
+        raise ValueError("Invalid hemispheres definition!\n" + supported)
+    return hemispheres
+
+
+def _check_views(views) -> list:
+    if not isinstance(views, list):
+        views = [views]
+    invalid_view = any([view not in VALID_VIEWS for view in views])
+    if invalid_view:
+        supported = "Supported views:\n" + str(VALID_VIEWS)
+        raise ValueError("Invalid view definition!\n" + supported)
+    return views
+
+
+def _colorbar_from_array(array, vmax,
+                         threshold, symmetric_cbar,
+                         kwargs,
+                         cmap='cold_hot'):
+    """ Generate a custom colorbar for an array.
+
+    Internal function used by plot_img_on_surf
+
+    array: Any 3D array
+
+    vmax : float, optional (default=None)
+        upper bound for plotting of stat_map values.
+
+    threshold : float, optional (default=None)
+        If None is given, the colorbar is not thresholded.
+        If a number is given, it is used to threshold the colorbar.
+        Absolute values lower than threshold are shown in gray.
+
+    symmetric_cbar : bool or 'auto', optional, default 'auto'
+         Specifies whether the colorbar should range from -vmax to vmax
+         or from vmin to vmax. Setting to 'auto' will select the latter
+         if the range of the whole image is either positive or negative.
+         Note: The colormap will always range from -vmax to vmax.
+
+    kwargs: extra arguments passed to _get_colorbar_and_data_ranges
+
+    cmap: str, optional (default='cold_hot')
+        the name of a matplotlib or nilearn colormap.
+    """
+
+    cbar_vmin, cbar_vmax, vmin, vmax = _get_colorbar_and_data_ranges(
```

Comment:
you can use cbar_vmin and cbar_vmax for symmetric_cbar to have an effect (or not expose a symmetric_cbar parameter)

Reply:
I'm not sure I understood exactly what you mean, please see the updated code and elaborate in case any changes are still required.

Reply:
I meant that as we are not taking it into account, we should probably remove the `symmetric_cbar` parameter (as I saw you have already done for `plot_img_on_surf`, thanks). 
My comment on `cbar_vmin` is that if we wanted to actually have a `symmetric_cbar` argument (we might add it in a future PR), we could get the necessary information from `cbar_vmin` and `cbar_vmax`

Reply:
I'm sorry but I'm still not sure I understand - `symmetric_cbar` is elsewhere used in `_colorbar_from_array()` and `plot_surf_stat_map()`. Would you like for me to remove it in both? They both call `_get_colorbar_and_data_ranges()`, which  is imported from the `img_plotting` module and requires this argument. If any one of the two calling functions omits the `symmetric_cbar` parameter we should probably also give it a default of "auto".

Reply:
when `_get_colorbar_and_data_ranges` is called with `symmetric_cbar=False` or "auto" resolves to `False`, it takes that into account by returning appropriate values for `cbar_vmin` and `cbar_vmax` . at the moment `_colorbar_from_array` ignores the values of `cbar_vmin` and `cbar_vmax`, so the value of `symmetric_cbar` has no effect, `_colorbar_from_array` always behaves as if `symmetric_cbar=True`. So my suggestion if (at least for now) to remove `symmetric_cbar` as a parameter from `_colorbar_from_array`'s signature, as it has no effect anyway, and always call `_get_colorbar_and_data_ranges(... , symmetric_cbar=True)`. A future PR can add this parameter and handle it corrrectly, but the new functionality is still useful without it.

regarding the call to `plot_surf_stat_map`, the value of `symmetric_cbar` doesn't matter because we call it with `cbar=False`: it does not add a colorbar here, instead `plot_img_on_surf` adds a colorbar for the whole figure separately.

so to summarize I suggest: no `symmetric_cbar` parameter for `plot_img_on_surf` nor `_colorbar_from_array`, and `_colorbar_from_array` calls `_get_colorbar_and_data_ranges` with `symmetric_cbar=True`

Reply:
Had to pass `symmetric_cbar=True` as a positional argument (avoiding changing anything in the `img_plotting` module) but otherwise I hope the updated code reflects what's you said.

Reply:
yes that is what I meant, thank you for bearing with me